packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3463,6 +3469,30 @@ export class ContainerRuntime
 		return result;
 	}
 
+	detachHead(): { merge(): void; pause(): Promise<void>; dispose(): void } {
```

Comment:
needs a better name, `enterStagingMode`, `startStaging`, etc

Reply:
do we want to some check the enforces this is only called once. the getBatchCheckpoints does some management, but it is transparent, so it might be better to error